lib/trento_web/openapi/v1/schema/user.ex

```diff
@@ -0,0 +1,118 @@
+defmodule TrentoWeb.OpenApi.V1.Schema.User do
+  @moduledoc false
+
+  require OpenApiSpex
+  alias OpenApiSpex.Schema
+
+  defmodule UserCreationRequest do
+    @moduledoc false
+
+    OpenApiSpex.schema(%{
+      title: "UserCreationRequest",
+      description: "Request body to create a user",
+      type: :object,
+      additionalProperties: false,
+      properties: %{
+        fullname: %Schema{type: :string, description: "User full name", nullable: false},
+        email: %Schema{type: :string, description: "User email", nullable: false, format: :email},
+        username: %Schema{type: :string, description: "User username", nullable: false},
+        enabled: %Schema{
+          type: :boolean,
+          description: "User enabled in the system",
+          nullable: false
+        },
+        password: %Schema{type: :string, description: "User new password", nullable: false},
+        password_confirmation: %Schema{
+          type: :string,
+          description: "User new password, should be the same as password field",
+          nullable: false
+        }
+      },
+      required: [:fullname, :email, :enabled, :password, :password_confirmation, :username]
+    })
+  end
+
+  defmodule UserUpdateRequest do
+    @moduledoc false
+
+    @schema %Schema{
+      title: "UserUpdateRequest",
+      description: "Request body to update a user",
+      type: :object,
+      additionalProperties: false,
+      properties: %{
+        fullname: %Schema{type: :string, description: "User full name", nullable: false},
+        email: %Schema{type: :string, description: "User email", nullable: false, format: :email},
+        enabled: %Schema{
+          type: :boolean,
+          description: "User enabled in the system",
+          nullable: false
+        },
+        password: %Schema{type: :string, description: "User new password", nullable: false},
+        password_confirmation: %Schema{
+          type: :string,
+          description: "User new password, should be the same as password field",
+          nullable: false
+        }
+      }
+    }
+
+    # see: https://github.com/open-api-spex/open_api_spex/issues/87
+    # This is an alternative way of defining schemas without having to deal to default values
+    # when the struct is converted to plain parameters in controller
+    # by default all the struct values have nil values, so we can't distinguish between a parameter not passed
+    # or a parameter passed explicitly nil.
+    # this is an update request and we have to distinguish in PATCH verb values passed or not passed.
+    # Having a value not passed as nil by default is not practical nor correct.
+    # this means that in the controller the body params are passed as map, without converting from the struct
+    # and we have everything casted and validated BUT without the hassle of cannot distinguish between passed
```

Comment:
I guess a typo in the tense:

and we have everything ~~casted~~ --> and we have everything **cast**

Reply:
done